### PR TITLE
new PR for pull-events-recorder doc, including latest changes

### PR DESCRIPTION
**Pull request by SharpRake**
## Type of change

My previous PR got all messed up, this is a new PR for the `events-recorder` example application (https://github.com/chainguard-dev/platform-examples/tree/main/event-recorder) companion doc. 

### What should this PR do?

Resolves https://github.com/chainguard-dev/internal/issues/3813

### Why are we making this change?

Our Events docs are out of date. We used to have three examples, now we only have one. Let's give our customers some different use cases of Events!

### What are the acceptance criteria? 

Should work, read smoothly, make sense, etc.

### How should this PR be tested?

This will require terraform, a GCP project, chainctl, plus a bunch of enabled APIs to get working. Please don't hesitate to ask me for help!

### CODE REVIEW

File: content/chainguard/administration/cloudevents/pull-events-recorder/index.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+Following that, run `terraform init` to initialize Terraform’s working directory.
+
+```sh
+terraform init
+```
+
+Then run `terraform plan`. This will produce a speculative execution plan that outlines what steps Terraform will take to create the resources defined in the files you set up in the last section.
+
+```sh
+terraform plan
```
[the remark this answers is not in the document]
Comment (sheesh) [reply content/chainguard/administration/cloudevents/pull-events-recorder/index.md:128]: Just documenting that Mark and I troubleshooted this error in the following ways:
* terraform init -upgrade
* restarting laptop
* trying in a brand new directory
Nothing worked. We're wondering if it is Mac issue.

**Review by sheesh — COMMENTED** [review]
Left a few comments before we were thwarted by the `terraform plan` error. Hope it works for Erika!

File: content/chainguard/administration/cloudevents/pull-events-recorder/index.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+The Terraform configuration you'll use in this guide will do all the work of setting up the example application's infrastructure and resources. Importantly, it creates a BigQuery dataset named `cloudevents_pull_event_recorder`, with a table named `dev_chainguard_registry_pull_v1`.
+
+The [`pull.schema.json` file](https://github.com/chainguard-dev/platform-examples/blob/main/event-recorder/iac/pull.schema.json) describes this dataset's schema, which contains fields describing the user who pulled the image, the image that was pulled, the time of the pull, and information about errors that occurred during the pull. This schema matches the description for the `registry:pul `event type described in the [Chainguard Events overview documentation](/chainguard/administration/cloudevents/events-reference/#service-registry---pull).
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/administration/cloudevents/pull-events-recorder/index.md:52]: yes!

File: content/chainguard/administration/cloudevents/pull-events-recorder/index.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+You can find all the code associated with this sample application in our [Platform Examples](https://github.com/chainguard-dev/platform-examples/tree/main/event-recorder) repository on GitHub. 
+
+To set up the sample application, you can create a Terraform configuration file that points to the application code in the Platform Examples repository and then apply it to set up the necessary resources. To begin, create a new directory to hold the configuration file and navigate into it.
+
+```sh
+mkdir ~/gcr-pull-data && cd $_
+```
+
+After navigating into the new directory you can begin creating a Terraform configuration named `main.tf`.
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/administration/cloudevents/pull-events-recorder/index.md:67]: not even, that sounds much cleaner

File: content/chainguard/administration/cloudevents/pull-events-recorder/index.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+This configuration will consist of a single module. For the purposes of this example, we will call it `event-recorder`. This module's `source` value will be the [`iac` folder](https://github.com/chainguard-dev/platform-examples/tree/main/event-recorder/iac) from the application code in the examples repository.
+
+```
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/administration/cloudevents/pull-events-recorder/index.md:71]: i like `input`

File: content/chainguard/administration/cloudevents/pull-events-recorder/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (erikaheidi) [reply content/chainguard/administration/cloudevents/pull-events-recorder/index.md:128]: I just tried it here, on Linux, and I got the same error on `terraform plan`:
```
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/event-recorder.cloudevent-broker/modules/regional-go-service/main.tf line 34, in resource "ko_build" "this":
│   34:   env         = each.value.source.env
│ 
│ An argument named "env" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/event-recorder.recorder/modules/regional-go-service/main.tf line 34, in resource "ko_build" "this":
│   34:   env         = each.value.source.env
│ 
│ An argument named "env" is not expected here.
╵
```

**Comment by ltagliaferri** [comment]
Currently blocked due to eng updates

**Pull request closed without merging**